datalake/src/test/java/com/sequenceiq/datalake/service/imagecatalog/ImageCatalogServiceTest.java

```diff
@@ -77,26 +91,88 @@ public void resourceCrnByResourceNameShouldBeResolvedByCore() {
     }
 
     @Test
-    public void testImageLookupByImageCatalogNameAndImageID() throws Exception {
-        ImageV4Response imageResponse = getImageResponse();
+    public void testGetImageResponseFromImageRequestWithPrewarmImage() throws Exception {
+        ImageV4Response imageResponse = getCdhImageResponse();
         ImagesV4Response imagesV4Response = new ImagesV4Response();
         imagesV4Response.setCdhImages(List.of(imageResponse));
 
         ImageSettingsV4Request imageSettingsV4Request = new ImageSettingsV4Request();
         imageSettingsV4Request.setCatalog(IMAGE_CATALOG_NAME);
         imageSettingsV4Request.setId(IMAGE_ID);
 
-        when(cloudbreakInternalCrnClient.withInternalCrn()).thenReturn(cloudbreakServiceCrnEndpoints);
-        when(cloudbreakServiceCrnEndpoints.imageCatalogV4Endpoint()).thenReturn(imageCatalogV4Endpoint);
         when(imageCatalogV4Endpoint.getImageByCatalogNameAndImageId(any(), eq(IMAGE_CATALOG_NAME), eq(IMAGE_ID), any())).thenReturn(imagesV4Response);
 
-        ThreadBasedUserCrnProvider.doAs(USER_CRN, () -> {
-            ImageV4Response actual = victim.getImageResponseFromImageRequest(imageSettingsV4Request, IMAGE_CATALOG_PLATFORM_AWS);
-            assertEquals(imageResponse, actual);
-        });
+        ImageV4Response actual = ThreadBasedUserCrnProvider.doAs(USER_CRN,
+                () -> victim.getImageResponseFromImageRequest(imageSettingsV4Request, IMAGE_CATALOG_PLATFORM_AWS));
+        assertEquals(imageResponse, actual);
```

Comment:
Note: `ImageV4Response` does not seem to override `equals()`, so this will essentially perform a reference equality check.